### DISCUSSION TITLE
Data type URIs for EDTF values

### DISCUSSION THREAD

**Discussion by nichtich** [periodo/edtf-ontology#14] [Q&A] [answered]
I assumed to find data type URIs for EDTF values to be used when `xsd:date`, `xsd:time`, `xsd:dateTime`, and `xsd:dateTimeStamp` are not applicable, e.g

~~~ttl
<a> dc:date "2023?"^^edtf:level1 .
<b> dc:date "‘Y-17E7"^^edtf:level2 .
~~~

Am I missing something?

**Comment by djradon** [comment, marked as answer]
It looks like the official datatype definitions are here: http://id.loc.gov/datatypes/EDTFScheme

They have three scheme members defined:

- http://id.loc.gov/datatypes/edtf/EDTF-level0
- http://id.loc.gov/datatypes/edtf/EDTF-level1
- http://id.loc.gov/datatypes/edtf/EDTF-level2

No substantial differences between those definitions. Interestingly, they all define their subjects as datatypes, classes, AND individuals.  

Maybe there's no level3 because it hasn't been formally adopted, see comments like https://github.com/citation-style-language/schema/issues/421#issuecomment-1147358318

**Comment by rybesh** [comment]
Hello!

There is also a 4th datatype defined as the union of the three levels: 

<http://id.loc.gov/datatypes/edtf/EDTF>

The range of [`inEDTFdatetime`](https://periodo.github.io/edtf-ontology/#inEDTFdatetime) is <http://id.loc.gov/datatypes/edtf/EDTF>, and the range of [`hasEDTFDateTimeSetDescription`](https://periodo.github.io/edtf-ontology/#hasEDTFdatetimesetdescription) is <http://id.loc.gov/datatypes/edtf/EDTF-level2>, since the concept of a “set description” only exists in level 2.

I didn’t bother explicitly putting datatypes on the EDTF literals in the examples because I was primarily interested in OWL inferencing over OWL-Time constructs generated from EDTF strings, and OWL tools basically ignore datatypes. But you are right that it would be more correct to do so, and might make the data easier to work with when using RDF-focused (rather than OWL-focused) tools.

I should also add that this repo has no connection to LoC or ISO, no official status whatsoever, just an experiment in modeling and inferencing over EDTF expressions.
